Helm cart and container

cunningpike opened the pull request:
Version 1.0.0 of a Helm chart and container for fediblockhole v0.4.2. Tested and working with flux kustomization in GKE.

jpwarren commented on chart/values.yaml:
```diff
@@ -0,0 +1,64 @@
+image:
+  repository: ghcr.io/cunningpike/fediblockhole
```
Why this repo source?

cunningpike replied:
It's where I have the built image right now - it should really be moved to your repo and that link updated...

jpwarren commented on chart/fediblockhole.conf.toml:
```diff
@@ -0,0 +1,67 @@
+#   higher level of authorization.
+# If `import_fields` are provided, only import these fields from the instance.
+#   Overrides the global `import_fields` setting.
+blocklist_instance_sources = [
+  # { domain = 'public.blocklist'}, # an instance with a public list of domain_blocks
+  # { domain = 'jorts.horse', token = '<a_different_token>' }, # user accessible block list
+  # { domain = 'eigenmagic.net', token = '<a_token_with_read_auth>', admin = true }, # admin access required
+]
```
Having a copy of the sample `fediblockhole.conf.toml` file here will require making the same changes in more than one place if the sample gets changed. Is there a way to make this easier to maintain?

cunningpike replied:
I *think* the chart needs a dummy file to create the part that pulls the value map to create the conf.toml file in /etc/default. When you deploy your own kustomization, it will actually pull the local configmap in your own flux repo. Open to ideas as to how to avoid dealing with changes in the sample file - that had occurred to me, but I thought we could deal with that when we rebuild the container each time the version increments?

cunningpike replied:
I don't think this part of the template works unless there is a file for it to pull - but it may not be necessary for deployment at all...

chart/templates/configmap-conf-toml.yaml

apiVersion: v1
kind: ConfigMap
metadata:
  name: {{ include "fediblockhole.fullname" . }}-conf-toml
  labels:
    {{- include "fediblockhole.labels" . | nindent 4 }}
data:
  {{ (.Files.Glob "fediblockhole.conf.toml").AsConfig | nindent 4 }}

jpwarren replied:
I just know that people will blindly use the sample provided if there is one. If it isn't maintained, it'll break stuff and they'll open bug reports when it does. Let's make it easy for people to get it right and hard to get it wrong, and really obvious when they do get it wrong _why_ it's wrong and how to fix it themselves.

cunningpike replied:
Let's remove that .yaml file from the template directory and see if it still deploys. If it does, I'd be in favor of removing it and placing an expanded version in the samples directory as you suggest. That's what the flux kustomization ends up using anyway, like this:

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: fediblockhole-conf-toml
  namespace: fediblockhole
data:
  fediblockhole.conf.toml: |-  
    # List of instances to read blocklists from.
    # If the instance makes its blocklist public, no authorization token is needed.
    #   Otherwise, `token` is a Bearer token authorised to read domain_blocks.
    # If `admin` = True, use the more detailed admin API, which requires a token with a 
    #   higher level of authorization.
    # If `import_fields` are provided, only import these fields from the instance.
    #   Overrides the global `import_fields` setting.
    blocklist_instance_sources = [
      # { domain = 'public.blocklist'}, # an instance with a public list of domain_blocks
      # { domain = 'jorts.horse', token = '<a_different_token>' }, # user accessible block list
      # { domain = 'eigenmagic.net', token = '<a_token_with_read_auth>', admin = true }, # admin access required
    ]
    ...
```

jpwarren commented on container/Dockerfile:
```diff
@@ -0,0 +1,17 @@
+RUN pip install fediblockhole
+
+USER 1001
+# Run the web service on container startup. Here we use the gunicorn
+# webserver, with one worker process and 8 threads.
+# For environments with multiple CPU cores, increase the number of workers
+# to be equal to the cores available.
```
This comment block doesn't seem to refer to what's happening here. Leftover from a cut+paste from elsewhere?

cunningpike replied:
Uh, probably...

jpwarren reviewed (commented):
A couple of questions I'd like clarified, but overall looks good!

Thanks for making the contribution!

cunningpike commented:
I have updated the chart and templates to version 1.1.0 and added the ability to specify local allow and block CSV files. These are mounted and loaded from configMaps, just as the conf.toml file is.
